game-core/src/main/java/games/strategy/triplea/delegate/BattleDelegate.java

```diff
@@ -677,146 +676,24 @@ private static void setupTerritoriesAbandonedToTheEnemy(
   }
 
   private void doScrambling() {
-    // first, figure out all the territories where scrambling units could scramble to
-    // then ask the defending player if they wish to scramble units there, and actually move the
-    // units there
     final GameData data = getData();
-    if (!Properties.getScrambleRulesInEffect(data)) {
```

Comment:
The previous logic was:
```
  final boolean toSbr = Properties.getCanScrambleIntoAirBattles(data);
  final Set<Territory> territoriesWithBattles =
        battleTracker.getPendingBattleSites().getNormalBattlesIncludingAirBattles();
    if (toSbr) {
      territoriesWithBattles.addAll(
          battleTracker.getPendingBattleSites().getStrategicBombingRaidsIncludingAirBattles());
    }
```

So my change is not changing anything from the logic. However, I agree the logic seems confusing. I'd rather not change it as part of this refactoring, though.

However, I actually think it's harmless, because it's being put into a set. So if we get some duplicate territories where air battles are happening, it won't be a problem, they'll be deduplicated by the set.

Reply:
For your first point, however, it means that every caller of ScrambeLogic would need to check getScrambleRulesInEffect(). This seems sub-optimal. In particular, just in this PR, I would need to add that if to the MovePanel code where I'm adding the new warning.

So I think the abstraction is actually helpful in simplifying the calling code.

Reply:
Wait, the original code block on `doScramble` had a:

```
if (!Properties.getScrambleRulesInEffect(data)) {
   return;
}
```

That appears to be missing now. Hence the above confusion I run into with this code block:
```
final Set<Territory> territoriesWithBattles =
        pendingBattleSites.getNormalBattlesIncludingAirBattles();
    if (Properties.getCanScrambleIntoAirBattles(data)) {
      territoriesWithBattles.addAll(
          pendingBattleSites.getStrategicBombingRaidsIncludingAirBattles());
    }

```

Before that made some sense, as we are certainly doing scrambling. Now it makes very little sense as we may or may not be scrambling.

Reply:
Ah, that's checked in ScrambleLogic now so I think it still has the same end result.

Reply:
Indeed, that's my point. A reader going through this will read through and try to understand a good bit of logic, to eventually find out the no-op portion comes later. In another way, this 'outer code' does not make sense until you can read through to the 'inner code'. Having to know about method implementations means the outer code is not good. Example:

```
void foo() {
  a();
  b();
  int c = d();
  e(c);
}
```
In the above, the methods make no sense, the abstraction is not helpful. If read the above, you have to stop at `a()`, find that, read it, find `b()`, read that, and then eventually you'll understand this whole method. It all may as well be inlined from a reader/maintenance point of view.

Compare to:
```
void foo() {
  if(a()) {
     b();
  }
  int c = d();
  e(c);    
}
```

In the above, if you know that `a` returns false, then you skip reading `b()`, this helps code maintenance scale well, it's in part about how much code you can skip reading because there is a separation of concerns. In the above, if `a` is false, you *know* that `b()` is not relevant and you can skip reading it.
```
We have that similar example here, there's a lot of outer code, if someone tries to read it and understand it without referencing any other methods, they won't be able to understand the flow properly. Having to reference method definitions to understand code is a maintenance anti-pattern.

Reply:
Ok, I'll make the change.

However, I still disagree. If two method calls always have to be used together, I've found it's better to combine them into one. It leads to less code and thus a reduction of complexity and the calling needing to know less implementation details. Now, each caller needs to know that there's this property thing that affects scrambling - rather than the logic being localized to the code specifically meant to abstract all of this away.

Reply:
Also, one other thing to consider - given ScrambeLogic takes battleTracker as a param, we can make it compute territoriesWithBattles internally. Anyway, we can do that in a later refactor.

Reply:
Thanks @asvitkine , it was either that or fix this code block:
```
final Set<Territory> territoriesWithBattles =
        pendingBattleSites.getNormalBattlesIncludingAirBattles();
    if (Properties.getCanScrambleIntoAirBattles(data)) {
      territoriesWithBattles.addAll(
          pendingBattleSites.getStrategicBombingRaidsIncludingAirBattles());
    }
```

Certainly we're operating under less than ideal circumstances here, so pretty sure there is no ideal solution.